pulsar-client/src/main/java/org/apache/pulsar/client/api/AuthenticationFactory.java

```diff
@@ -41,17 +41,18 @@
      */
     public static final Authentication create(String authPluginClassName, String authParamsString)
             throws UnsupportedAuthenticationException {
-        Map<String, String> authParams = new HashMap<String, String>();
-        if (isNotBlank(authParamsString)) {
-            String[] params = authParamsString.split(",");
-            for (String p : params) {
-                String[] kv = p.split(":");
-                if (kv.length == 2) {
-                    authParams.put(kv[0], kv[1]);
-                }
+        try {
+            if (isNotBlank(authPluginClassName)) {
+                Class<?> authClass = Class.forName(authPluginClassName);
+                Authentication auth = (Authentication) authClass.newInstance();
+                auth.configure(authParamsString);
```

Comment:
a quick thought on implementation:
Rather adding `void configure(String authParams)` method in `Authentication.java` and doing parsing, instead can we do parsing here in `AuthenticationFactory`?

1. First try to parse given auth-param into json and prepare `Map<String,String>` from json and call configure(map)
`Map<String, String> authParamsMap = jsonMapper.readValue(authParamsString, new TypeReference<HashMap<String, String>>() {});`
2. If fails then try to parse in existing way with `:` and `;` delimiters, and prepare config-map

Reply:
I think rdhabalia's suggestion is good enough since what we want to do here is to allow String containing ":" or "," (such as URL) as a parameter.

Introducing new interface sounds nice, however, I think we should do that when we really need plugin-specific parsing.

Reply:
Well, the new interface may be YAGNI. However, I still don't understand why you guys want limitations.

- Maybe `Map<String, String>` is enough, maybe not. Who knows?
- colon-comma, JSON, YAML... What's next? Can we auto-detect all formats?
- When we finally agreed on having new interface, is it still easy to migrate? How many plugins exists?

What will we lose instead of freedom if we introduce the new one now?

In my opinion, we should maximize API capability and do smart things on top of it.

Reply:
> What will we lose instead of freedom if we introduce the new one now?

Sure, I will be fine with new api as well. 👍